Added a describe_features function to the SASAFeaturizer.

dr-nate opened the pull request:
- [x] Implement feature / fix bug
 - [x] Add tests
 - [ ] Update changelog

Added a `describe_features` function to the SASAFeaturizer. Contingent on this PR to MDTraj also being accepted https://github.com/mdtraj/mdtraj/pull/1174. Not a huge addition but it's nice to have. Let me know if this is significant enough to add to the changelog. 

Also, I have tested it on real data, but didn't add any tests. Let me know if you want those too.

msultan commented on msmbuilder/featurizer/featurizer.py:
```diff
@@ -884,6 +884,55 @@ def __init__(self, mode='residue', **kwargs):
+            resseqs = [top.residue(ri).resSeq for ri in resids]
+            resnames = [top.residue(ri).name for ri in resids]
+            # Not sure if there's a better way to get the indices of atoms in a residue
+            atoms_in_res = [res.atoms for res in top.residues]
```
This looks good to me and i cant think of a fancy list comprehensive way around this. Maybe @mpharrigan knows.

msultan commented:
I think a test would still be nice. Take a look at some of the randomized tests that i wrote [here](https://github.com/msmbuilder/msmbuilder/blob/master/msmbuilder/tests/test_feature_descriptor.py) for taking a random frame and calculating a random feature index to make sure they match up.

msultan commented on msmbuilder/featurizer/featurizer.py:
```diff
@@ -884,6 +884,55 @@ def __init__(self, mode='residue', **kwargs):
+                aind_tuples.append([])
+                # For each atom in the residues, append it's index
+                for y in x:
+                    aind_tuples[i].append(y.index)
```
this could be 
```
aind_tuples.append([atom.index for atom in x])
```

msultan commented:
Thanks for taking this on @dr-nate !

dr-nate commented:
Ok. I didn't realize there was already test for the other `describe_features` functions. I've added them now and cleaned up the other code. If/when the MDTraj PR is accepted, this one can be accepted.

msultan commented:
@mpharrigan could you take a look at the Travis error, I don't really understand whats going on here.

mpharrigan commented:
Attempting to fix in https://github.com/msmbuilder/msmbuilder/pull/915
